tsdb/engine/tsm1/compact_test.go

```diff
@@ -562,64 +562,94 @@ func TestCompactor_CompactFull_TombstonedMultipleRanges(t *testing.T) {
 	}
 }
 
-// Tests that a single TSM file can be read and iterated over
-func TestTSMKeyIterator_Single(t *testing.T) {
+// Ensures that a compaction will properly rollover to a new file when the
+// max keys per blocks is exceeded
+func TestCompactor_CompactFull_MaxKeys(t *testing.T) {
+	// This test creates a lot of data and causes timeout failures for these envs
+	if testing.Short() || os.Getenv("CIRCLECI") != "" || os.Getenv("APPVEYOR") != "" || os.Getenv("GORACE") != "" {
+		t.Skip("Skipping max keys compaction test")
+	}
```

Comment:
CircleCI and Appveyor both set the `CI` environment variable.

https://circleci.com/docs/environment-variables/
https://www.appveyor.com/docs/environment-variables